Refactor Git panel styling & status colors for consistency

jakucermak opened the pull request:
Closes #26847

Release Notes:

- Updated Git panel background to use panel_background instead of ElevationIndex::Surface.bg(cx) for consistency with other panels.
- Removed redundant GitStatusColors struct from status.rs and refactored to use existing theme colors.
- Adjusted Color enum mappings in color.rs to reference version_control_* colors instead of status() for better alignment with the theme system.
- Cleaned up unused or redundant code.

jakucermak committed:
Use panel_background color for Git panel background

jakucermak committed:
Remove GitStatusColors struct and update color references

jakucermak committed:
Merge branch 'zed-industries:main' into main

maxbrunsfeld commented:
Can you add a before and after screenshot?

jakucermak commented:
## File status color in project panel and git gutter:
### Before:
<img width="706" alt="image" src="https://github.com/user-attachments/assets/e9e7668b-0c2b-4d39-9600-7408f0cb0f2d" />

### After: 
<img width="706" alt="image" src="https://github.com/user-attachments/assets/fe99428b-64a9-4a7f-87e7-44597f735047" />

## Git panel background color:
### Before:
#### Git panel
<img width="483" alt="image" src="https://github.com/user-attachments/assets/ca441d74-acb0-483f-95ae-62d5013b5d22" />

#### Project panel:
<img width="483" alt="image" src="https://github.com/user-attachments/assets/f2e6025f-5f5b-4651-83cc-070b303c1f0d" />

### After
#### Git panel
<img width="483" alt="image" src="https://github.com/user-attachments/assets/b5c1df85-f7a4-4921-8b87-f79acec5a6c0" />

#### Project panel:
<img width="483" alt="image" src="https://github.com/user-attachments/assets/d4d1897b-191c-43b6-abcf-62b178ef8a01" />

iamnbutler commented on crates/ui/src/styles/color.rs:
This we don't want – We should add VersionControl{Variant} to color, is there are other current and future usages of generic Created, Modified, etc that we want to be independently theme-able.

Otherwise looks good!

iamnbutler commented:
Hey, thanks for the contribution!

Other than updating that one change to the Color struct this looks good. @me when that is fixed and I can get this merged.

jakucermak committed:
Refactor color theming: Introduce VersionControl{Variant}

- Added distinct `VersionControl{Variant}` colors in `Color` enum
- Updated git panel and status label logic to use
`VersionControl{Variant}`
- Ensured version control colors are independently theme-able
- Removed redundant mappings between generic status colors and version
control

This change allows `Created`, `Modified`, etc., to remain generic while
version control-specific statuses have dedicated colors.

jakucermak committed:
Merge branch 'main' of github.com:jakucermak/zed

jakucermak committed:
Merge branch 'main' into main

iamnbutler reviewed (approved):
lgtm :)

shellscape commented:
After this update my colors went from pleasant and pleasing to wtf. This is how they look now, and it aint cool.

<img width="207" alt="Screenshot 2025-03-21 at 12 21 43 PM" src="https://github.com/user-attachments/assets/fb6900c4-22fd-48f3-8c2e-f7a4c3e2a3ab" />
